modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectLayoutLocalServiceImpl.java

```diff
@@ -393,31 +444,51 @@ private ObjectLayoutTab _addObjectLayoutTab(
 	}
 
 	private List<ObjectLayoutTab> _addObjectLayoutTabs(
-			User user, long objectDefinitionId, long objectLayoutId,
+			User user, long objectDefinitionId, ObjectLayout objectLayout,
 			List<ObjectLayoutTab> objectLayoutTabs)
 		throws PortalException {
 
-		return TransformUtil.unsafeTransform(
+		objectLayoutTabs = TransformUtil.unsafeTransform(
 			objectLayoutTabs,
 			objectLayoutTab -> _addObjectLayoutTab(
-				user, objectDefinitionId, objectLayoutId,
+				user, objectDefinitionId, objectLayout.getObjectLayoutId(),
 				objectLayoutTab.getObjectRelationshipId(),
 				objectLayoutTab.getNameMap(), objectLayoutTab.getPriority(),
 				objectLayoutTab.getObjectLayoutBoxes()));
+
+		if (objectLayout.isDefaultObjectLayout()) {
+			registryObjectLayoutTabScreenNavigationCategory(
+				_objectDefinitionPersistence.fetchByPrimaryKey(
+					objectDefinitionId),
+				objectLayoutTabs);
+		}
+
+		return objectLayoutTabs;
 	}
 
 	private void _deleteObjectLayoutBoxes(
 		List<ObjectLayoutTab> objectLayoutTabs) {
 
 		for (ObjectLayoutTab objectLayoutTab : objectLayoutTabs) {
-			List<ObjectLayoutBox> objectLayoutBoxes =
+			_deleteObjectLayoutRows(
 				_objectLayoutBoxPersistence.findByObjectLayoutTabId(
-					objectLayoutTab.getObjectLayoutTabId());
+					objectLayoutTab.getObjectLayoutTabId()));
 
 			_objectLayoutBoxPersistence.removeByObjectLayoutTabId(
 				objectLayoutTab.getObjectLayoutTabId());
 
-			_deleteObjectLayoutRows(objectLayoutBoxes);
+			ServiceRegistration<?> serviceRegistration =
+				_serviceRegistrationMap.get(
+					_getServiceRegistrationMapKey(objectLayoutTab));
+
+			if (serviceRegistration == null) {
+				return;
+			}
+
+			serviceRegistration.unregister();
```

Comment:
@MarinhoFeliphe Why we're unregistering tabs components when removing ObjectLayoutBoxes (instead of when deleting tabs itself) ?

Reply:
@gabrielwas In this method we're removing object layout boxes, 

object layout tabs are been removed before ([here](https://github.com/liferay/liferay-portal/blob/6b26ec819e30d11793131d8322c24a2cf81a2d9e/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectLayoutLocalServiceImpl.java#L524)),

I just used the `for` that is performed to delete boxes to unregister object layout tabs.